core.js

```diff
@@ -1355,6 +1355,23 @@ async function _fromTokenizer(tokenizer) {
 			}
 		}
 	}
+
+	// Check for Asar
```

Comment:
This can be simplified down to:

```js
	// Check for Asar
	try {
		const headerSize = buffer.readUInt32LE(12);
		const header = buffer.slice(16, headerSize + 16).toString();
		JSON.parse(header);
		return {
			ext: 'asar',
			mime: 'application/asar'
		};
	} catch (_) {}
```

Reply:
Thanks. Merged your suggestion.

Reply:
If would prefer to first make a assessment (an educated guess) on binary level if a file is a JSON / Pickle file before parsing it as JSON. To prevent unnecessary parsing of JSON and prevent errors.

Reply:
In case this sample is Asar file, is the header indeed always exact 16 bytes?

Reply:
Thanks for review.
I'm sorry, I don't have knowledge to detect type as binary level.

> In case this sample is Asar file, is the header indeed always exact 16 bytes?

Worked in multiple Asar files when I tried.
But, I didn't find an official evidence for this thing.

Do anyone have an idea?

Reply:
To me it looks like [asar](https://github.com/electron/asar#format) is using [Pickle](https://chromium.googlesource.com/chromium/src/+/master/base/pickle.h) to serialize.

> Worked in multiple Asar files when I tried.
> But, I didn't find an official evidence for this thing.

There must be some clue this supposed to work. If you cannot find a written explanation, you could try to derive it from the source code.: [pickle.h](https://chromium.googlesource.com/chromium/src/+/master/base/pickle.h), [pickle.cc](https://chromium.googlesource.com/chromium/src/+/master/base/pickle.cc). To me it looks like Pickle is starting with header portion of 4 bytes (32 bit).

If the length is indeed fixed for Asar (I don't think this is the case for any Pickle format), you may use the length withing the 4 byte portion as an initial signature. The `ext` in the JSON header part may be another one.

Reply:
Thanks your advices.
I am investigating. 
It still takes a while.

Reply:
@Borewit 

I found clues from `electron/asar` source code and wrote comments about this.
I also added a checking binary process, but this method may be poor.
Could you review again?

CC: @Richienb

Reply:
I will, I am sorry I let you wait.